Some small enhancements

horiavmuntean opened the pull request:
Some small enhancements in one commit

Commit:
[1] make the project work in windows (normalize keystore path)
[2] make index.html load hostname in URL
[3] change the logging framework so that program can be run with mvn compile exec:java -Dorg.eclipse.jetty.LEVEL=DEBUG -Dexec.mainClass="org.example.Main"
[4] new self-signed keystore.p12 with CN=demo instead of localhost because Firefox can't load the page with localhost; demo must be put in browsers hosts file

lachlan-roberts commented on src/main/java/org/example/Main.java:
```diff
@@ -72,7 +73,7 @@ public static void main(String[] args) throws Exception
 
         server.start();
-        System.err.println("https://localhost:8443/");
+        System.err.println("https://demo:8443/");
```
This doesn't work by default. Probably best to leave as localhost.

horiavmuntean replied:
The initial reason for this change was that Firefox does not work with an self-signed certificate having only CN=localhost, it will display Error code: SEC_ERROR_INADEQUATE_KEY_USAGE and there is no way to go around this. This is why I changed the keystore.p12 also with CN=demo.

But now I generated a self-signed certificate with CN=localhost but added subjectAltName as well and now it seems Firefox is loading https://localhost:8443 wit no problems.

The keystore can be generated as:

```
openssl req -newkey rsa:2048 -nodes -keyout key.pem -x509 -days 3650 -out certificate.pem \
  -subj "/CN=localhost" \
  -addext "subjectAltName=DNS:demo"
  
openssl pkcs12 -inkey key.pem -in certificate.pem -export -out keystore.p12
```

Should I make another PR with these changes ?

Thanks

lachlan-roberts replied:
I don't mind as long as it can still be used without external configuration.

You could add another line for `https://demo:8443/` with a message how to configure this.

Commit:
Rollback to localhost, but changed keystore.p12 to be generated with subjectAltName:

openssl req -newkey rsa:2048 -nodes -keyout key.pem -x509 -days 3650 -out certificate.pem \ -subj "/CN=localhost" \ -addext "subjectAltName=DNS:demo"
openssl pkcs12 -inkey key.pem -in certificate.pem -export -out keystore.p12

Now FF will load https://localhost:8443
